sdk/docs/manually-written/sdk/sdlc-howtos/applications/develop/contracts-and-transactions-in-java.rst

```diff
@@ -0,0 +1,158 @@
+
+    // Create a client object to access services on the ledger.
+    DamlLedgerClient client = DamlLedgerClient.newBuilder(ledgerhost, ledgerport).build();
+
+    // Connects to the ledger and runs initial validation.
+    client.connect();
+
+.. _howto-applications-work-with-contracts-java-authorization:
+
+Perform authorization
+^^^^^^^^^^^^^^^^^^^^^
+
+Some ledgers enforce authorization and require to send an access token along with each request. For more details on
+authorization, read the :ref:`Authorization <authorization>` overview.
+
+To use the same token for all Ledger API requests, use the ``withAccessToken`` method of the ``DamlLedgerClient`` builder.
+This allows you to not pass a token explicitly for every call.
+
+If your application is long-lived and your tokens are bound to expire, reload the necessary token when needed and pass
+it explicitly for every call. Every client method has an overload that allows a token to be passed, as in the following example:
+
+.. code-block:: java
+
+   stateClient.getLedgerEnd(); // Uses the token specified when constructing the client
+   stateClient.getLedgerEnd(accessToken); // Override the token for this call exclusively
+
+If you are communicating with a ledger that verifies authorization it is very important to secure the communication
+channel to prevent your tokens to be exposed to man-in-the-middle attacks. The next chapter describes how to enable TLS.
+
+.. _howto-applications-work-with-contracts-java-connect-securely:
+
+Connect securely
+^^^^^^^^^^^^^^^^
+
+The builders created by ``DamlLedgerClient.newBuilder`` default to a plaintext connection.
+The Java bindings library lets you connect to a ledger via a secure connection. To do so, invoke ``withSslContext`` and
+pass an ``SslContext``.
+
+.. warning::
+
+  Use the default plaintext connection only when connecting to a locally running ledger for development purposes.
+
+Secure connections to a ledger must be configured to use client authentication certificates, which can be provided by a ledger operator.
+
+For information on how to set up an ``SslContext`` with the provided certificates for client authentication, please consult the gRPC documentation on
+`TLS with OpenSSL <https://github.com/grpc/grpc-java/blob/master/SECURITY.md#tls-with-openssl>`_ as well as the
+`HelloWorldClientTls <https://github.com/grpc/grpc-java/blob/70b1b1696a258ffe042c7124217e3a7894821444/examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java#L46-L57>`_ example of the ``grpc-java`` project.
+
+Advanced connection settings
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Sometimes the default settings for gRPC connections or channels are not suitable. For such situations, create a custom
+`NettyChannelBuilder <https://grpc.github.io/grpc-java/javadoc/io/grpc/netty/NettyChannelBuilder.html>`_ object and
+pass the it to the ``newBuilder`` static method defined over `DamlLedgerClient <javadocs/com/daml/ledger/rxjava/DamlLedgerClient.html>`_.
```

Comment:
Is this link safe? @pawelulewicz-da 
The same already existed in https://github.com/digital-asset/daml/blob/2ca992544ab030abdaabe7ec11a0b5d750562493/sdk/docs/manually-written/sdk/sdlc-howtos/applications/integrate/bindings-java/index.rst?plain=1#L59.

Reply:
when locally deploying the daml docs is broken ofcourse